### DISCUSSION TITLE
using hidden_field

### DISCUSSION THREAD

**Discussion by wdiechmann** [yippee-fun/phlex#502] [Help] [answered]
How do I provide for the adapter sufficiently?

```
module Views
  #
  # the date component is a wrapper for the date_field form input field
  #
  class Components::Form::HiddenField < Phlex::HTML
    include Phlex::Rails::Helpers::HiddenField

    def initialize( form: nil, field: nil, value: nil )
      @form = form
      @field = field
      @value = value
    end

    def template()
      @form.hidden_field( @field, value: @value)
    end
  end
end

```

renders:

```
<input autocomplete="off" type="hidden" name="account_id[{:value=>3}]" id="account_id_{:value=>3}">
```

whereas I'd like it to render:

```
<input value="3" autocomplete="off" type="hidden" name="asset[account_id]" id="asset_account_id">
```

(wrapping the form (in this case asset) nicely around the field attribute - as Rails does out-of-the-box so to speak)

**Comment by wdiechmann** [comment]
I've solved this issue - not how I'd like it to work but ...

```
module Views
  #
  # the date component is a wrapper for the date_field form input field
  #
  class Components::Form::HiddenField < Phlex::HTML
    include Phlex::Rails::Helpers::HiddenField

    def initialize( resource: nil, field: nil, value: nil )
      @resource = resource.class.to_s.underscore
      @field = field
      @value = value
    end

    def template()
      hidden_field( @resource, @field, value: @value)
    end
  end
end

```

I don't get to enjoy the form_builder niceties but see the rendering I like 

```
<input value="3" autocomplete="off" type="hidden" name="event[account_id]" id="event_account_id">
```

**Comment by joeldrapper** [comment, marked as answer]
The `HiddenField` adapter corresponds to this view helper https://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html#method-i-hidden_field_tag

You should never need to use that adapter because it's better to work with the yielded form builder from the `form_with` or `form_for` adapters.

The `FormWith` adapter yields an adapted Rails FormBuilder that corresponds to this https://api.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html

That means you can call methods on the yielded form builder like this:

```ruby
class Foo < Phlex::HTML
  include Phlex::Rails::Hepers::FormWith

  def template
    form_with(model: Post.new) do |f|
      f.hidden_field :bar
    end
  end
end
```

This is the equivalent of this in ERB and produces the same output.

```erb
<%= form_with(model: Post.new) do |f| %>
  <%= f.hidden_field :bar %>
<% end %>
```
